### PR TITLE
test: Database startup error when there are multiple active generations

### LINKED ISSUE #2196: Changes to server startup to support database deletion
Design from #1881:

Check for tombstone files within generation subdirectories before adding a database to the catalog
  - [ ] If there are more than one generation subdirectories missing tombstone files, this is a bad state. Mark the database as being in an error state and continue startup.
  - [x] If there aren't any generation subdirectories missing tombstone files, ~~don't add that database to the catalog at all.~~ Add it to the catalog in a state that indicates there are no active generation directories. Done in 6d0959fbc.

### PR DESCRIPTION
**Pull request by carols10cents**
Fixes #2196.

Most of the server startup work was done in https://github.com/influxdata/influxdb_iox/commit/6d0959fbc3c0bfb7420caa9e92b0ba7e250f7d6b but this adds a test for the expected error state of a database when there's more than one generation directory that doesn't have a tombstone file.

### CODE REVIEW

**Review by NGA-TRAN — APPROVED** [review]
LGTM. I am wondering whether we plan to add in existing tests that with this parttern:
1. Create DB
2. Use that DB for some loads and queries
3. Delete DB
4. Use it and get error 
5. Create the same DB again
6. Use it again (loads and queries)

File: server/src/lib.rs [hunk 1]
```diff
@@ -2050,6 +2054,83 @@ mod tests {
...
+            .flatten()
+            .collect();
+
+        // Delete all tombstone files
+        for file in database_files {
+            let parsed: DirsAndFileName = file.clone().into();
+            if parsed.file_name.unwrap().to_string() == "DELETED" {
+                application.object_store().delete(&file).await.unwrap();
+            }
+        }
```
Comment (crepererum) [inline server/src/lib.rs:2107]:
```suggestion
        // Delete all tombstone files
        let mut deleted_something = false;
        for file in database_files {
            let parsed: DirsAndFileName = file.clone().into();
            if parsed.file_name.unwrap().to_string() == "DELETED" {
                application.object_store().delete(&file).await.unwrap();
                deleted_something = true;
            }
        }
        assert!(deleted_something);
```

If the tombstone pattern ever changes, we should ensure that this test still deletes them. I'm using a similar pattern when messing around w/ the catalog. While for sure it's not perfect it's at least some safety net.

**Review by crepererum — APPROVED** [review]
Apart from the additional safety net I would like to see, that's a solid test :+1:

**Comment by carols10cents** [comment]
@NGA-TRAN Great idea! I filed https://github.com/influxdata/influxdb_iox/issues/2454 to track that :) Thanks!

**Pull request merged**